tests/test_dataset.py

```diff
@@ -0,0 +1,520 @@
+
+    print(written_data.dtype)
+
+    assert np.array_equal(data, written_data)
+
+
+def test_wk_write_multi_channel_uint16():
+    dataset_path = "../testoutput/wk_multichannel/"
+    delete_dir(dataset_path)
+
+    ds_tiff = WKDataset.create(dataset_path, scale=(1, 1, 1))
+    mag = ds_tiff.add_layer("color", "color", num_channels=3, dtype=np.uint16).add_mag(
+        "1"
+    )
+
+    # 10 images (z-layers), each 250x250, dtype=np.uint16
+    data = np.zeros((3, 250, 250, 10), dtype=np.uint16)
+    for h in range(10):
+        for i in range(250):
+            for j in range(250):
+                data[0, i, j, h] = i * 256
+                data[1, i, j, h] = j * 256
+                data[2, i, j, h] = 100 * 256
+
+    mag.write(data)
+    written_data = mag.read(size=(250, 250, 10))
+
+    print(written_data.dtype)
+
+    assert np.array_equal(data, written_data)
+
+
+def test_wkw_empty_read():
+    filename = "../testoutput/Empty_WKDataset"
+    delete_dir(filename)
+
+    mag = (
+        WKDataset.create(filename, scale=(1, 1, 1))
+        .add_layer("color", "color")
+        .add_mag("1")
+    )
+    data = mag.read(size=(0, 0, 0), offset=(1, 1, 1))
+
+    assert data.shape == (1, 0, 0, 0)
+
+
+def test_tiff_empty_read():
+    filename = "../testoutput/Empty_TiffDataset"
+    delete_dir(filename)
+
+    mag = (
+        TiffDataset.create(filename, scale=(1, 1, 1))
+        .add_layer("color", "color")
+        .add_mag("1")
+    )
+    data = mag.read(size=(0, 0, 0), offset=(1, 1, 1))
+
+    assert data.shape == (1, 0, 0, 0)
+
+
+def test_tiff_read_padded_data():
+    filename = "../testoutput/Empty_TiffDataset"
+    delete_dir(filename)
+
+    mag = (
+        TiffDataset.create(filename, scale=(1, 1, 1))
+        .add_layer("color", "color", num_channels=3)
+        .add_mag("1")
+    )
+    # there are no tiffs yet, however, this should not fail but pad the data with zeros
+    data = mag.read(size=(10, 10, 10))
+
+    assert data.shape == (3, 10, 10, 10)
+    assert np.array_equal(data, np.zeros((3, 10, 10, 10)))
+
+
+def test_wk_read_padded_data():
+    filename = "../testoutput/Empty_WKDataset"
+    delete_dir(filename)
+
+    mag = (
+        WKDataset.create(filename, scale=(1, 1, 1))
+        .add_layer("color", "color", num_channels=3)
+        .add_mag("1")
+    )
+    # there are no tiffs yet, however, this should not fail but pad the data with zeros
+    data = mag.read(size=(10, 10, 10))
+
+    assert data.shape == (3, 10, 10, 10)
+    assert np.array_equal(data, np.zeros((3, 10, 10, 10)))
+
+
+def test_read_and_write_of_properties():
+    destination_path = "../testoutput/read_write_properties/"
+    delete_dir(destination_path)
+    source_file_name = "../testdata/simple_tiff_dataset/datasource-properties.json"
+    destination_file_name = destination_path + "datasource-properties.json"
+
+    imported_properties = TiffProperties.from_json(source_file_name)
+    imported_properties.path = destination_file_name
```

Comment:
Is this a typical usage of the `TiffProperties` class? When would you set the `path` property?

Reply:
No, this is not the typical usage of the `TiffProperties`. The developer does not need to do this, because this is handled by the Dataset. However, I do it explicitly in this test, to test if reading and writing properties work.